[launch-darkly-server] Duplicate symbol when linked with cryptopp

**Describe the bug**
SHA1 is duplicated in static library builds

**Environment**
- OS: LINUX
- Compiler: gcc

**To Reproduce**
Steps to reproduce the behavior:
reproducer is here - 
https://github.com/launchdarkly/c-server-sdk/issues/77

**Expected behavior**
It does not produce a duplicate symbol

**Failure logs**
In linked issue

**Additional context**

I've fixed this on my side with the following two patches

clibs patch
```
diff --git a/sha1.c b/sha1.c
index 76cd6ca..b30617c 100644
--- a/sha1.c
+++ b/sha1.c
@@ -279,7 +279,7 @@ void SHA1Final(
     memset(&finalcount, '\0', sizeof(finalcount));
 }
 
-void SHA1(
+void SHA1_CLIBS(
     char *hash_out,
     const char *str,
     uint32_t len)
diff --git a/sha1.h b/sha1.h
index f492009..b7bf5a0 100644
--- a/sha1.h
+++ b/sha1.h
@@ -40,7 +40,7 @@ void SHA1Final(
     SHA1_CTX * context
     );
 
-void SHA1(
+void SHA1_CLIBS(
     char *hash_out,
     const char *str,
     uint32_t len);
```

launch darkly patch

```
diff --git a/src/evaluate.c b/src/evaluate.c
index 8c4c664..c1adceb 100644
--- a/src/evaluate.c
+++ b/src/evaluate.c
@@ -1246,7 +1246,7 @@ LDi_bucketUser(
             char        digest[21], encoded[17];
             const float longScale = 1152921504606846975.0;
 
-            SHA1(digest, raw, strlen(raw));
+            SHA1_CLIBS(digest, raw, strlen(raw));
 
             /* encodes to hex, and shortens, 16 characters in hex 8 bytes */
             status = hexify(

```

these are for 2.9.2 and my corresponding portfile.cmake is

```
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO launchdarkly/c-server-sdk
    REF 2.9.2
    SHA512 8d32087728f5c3939b5898735541b01fec35e1555439719c44494e30cce603768d40912b4da22dabc967fa2ec889b1b68864003a05d121b57a13b86eb59499d5
    PATCHES 
        ${CMAKE_CURRENT_LIST_DIR}/sha1.patch   
)

vcpkg_from_github(
    OUT_SOURCE_PATH HEXIFY_SOURCE_DIR
    REPO pepaslabs/hexify.c
    REF f823bd619f73584a75829cc1e44a532f5e09336e
    SHA512 fdfd3877874cb5b3e506d791c08840b156ba6905cee520adc225755f7ca768e54a2efa4d05cbef72d275ca2596e1a4d8e4fbb254f9cc4188c31a41b9904479bc
    HEAD_REF master
    PATCHES
        ${SOURCE_PATH}/patches/hexify.patch
)

vcpkg_from_github(
    OUT_SOURCE_PATH SEMVER_SOURCE_DIR
    REPO h2non/semver.c
    REF bd1db234a68f305ed10268bd023df1ad672061d7
    SHA512 29c7ab45e6550977bb6c74ebddfff440559a6e6494b701fc69a815912e6d683e5f4b4dfe17c98a892e8a82766f33c83edbe11e973579bd1d2175384cbaadd731
    HEAD_REF master
    PATCHES 
        ${SOURCE_PATH}/patches/semver.patch
)

vcpkg_from_github(
    OUT_SOURCE_PATH SHA1_SOURCE_DIR
    REPO clibs/sha1
    REF d9ae30f34095107ece9dceb224839f0dc2f9c1c7
    SHA512 2ed0ad05a6afb34dab4d21fada57369718e3a48113eea3e55e291ecaea36604c428ba9d2b702e9d01a8703eb06aa5d42895b8d197959ca8df18605ec52b1745e
    HEAD_REF master
    PATCHES 
        ${CMAKE_CURRENT_LIST_DIR}/clibs.patch
)

vcpkg_from_github(
    OUT_SOURCE_PATH TIMESTAMP_SOURCE_DIR
    REPO chansen/c-timestamp
    REF b205c407ae6680d23d74359ac00444b80989792f
    SHA512 497a1766b58e6e1c5ff2edf4fd6ff5f1065c4bdac60767988a5da03f93b0724ef582240aa015f3ab724200c249fe98072c67efbfc90e54d986a42212b43030ea
    HEAD_REF master
    PATCHES 
        ${SOURCE_PATH}/patches/timestamp.patch
)

vcpkg_from_github(
    OUT_SOURCE_PATH UTHASH_SOURCE_DIR
    REPO troydhanson/uthash
    REF v2.3.0
    SHA512 3b01f1074790fb242900411cb16eb82c1a9afcf58e3196a0f4611d9d7ef94690ad38c0a500e7783d3efa20328aa8d6ab14f246be63b3b3d385502ba2b6b2a294
    HEAD_REF master
   )

vcpkg_cmake_configure(
    SOURCE_PATH ${SOURCE_PATH}
    OPTIONS
        ${FEATURE_OPTIONS}
        -DBUILD_TESTING=OFF
        -DFETCHCONTENT_SOURCE_DIR_HEXIFY=${HEXIFY_SOURCE_DIR}
        -DFETCHCONTENT_SOURCE_DIR_SEMVER=${SEMVER_SOURCE_DIR}
        -DFETCHCONTENT_SOURCE_DIR_SHA1=${SHA1_SOURCE_DIR}
        -DFETCHCONTENT_SOURCE_DIR_TIMESTAMP=${TIMESTAMP_SOURCE_DIR}
        -DFETCHCONTENT_SOURCE_DIR_UTHASH=${UTHASH_SOURCE_DIR}
        -DSKIP_DATABASE_TESTS=OFF

)

vcpkg_cmake_install()
vcpkg_cmake_config_fixup(
	CONFIG_PATH lib/cmake/ldserverapi
)

message (STATUS "TEST - ${CMAKE_CURRENT_LIST_DIR}")
set(shareDir "${CURRENT_PACKAGES_DIR}/share/${PORT}")
file(COPY ${CMAKE_CURRENT_LIST_DIR}/launch-darkly-server-config.cmake DESTINATION ${shareDir})
file(RENAME ${shareDir}/ldserverapiTargets.cmake ${shareDir}/ldserverapi-targets.cmake)

file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")

file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

This report proposes a solution but is brief on the problem.
~~~
: && /usr/bin/c++     CMakeFiles/test.dir/main.cxx.o  -o test  vcpkg_installed/x64-linux/debug/lib/libldserverapi.a  vcpkg_installed/x64-linux/debug/lib/libcryptopp.a  vcpkg_installed/x64-linux/lib/libcurl.a  vcpkg_installed/x64-linux/lib/libssl.a  vcpkg_installed/x64-linux/lib/libcrypto.a  -ldl  vcpkg_installed/x64-linux/lib/libssl.a  vcpkg_installed/x64-linux/lib/libcrypto.a  -ldl  vcpkg_installed/x64-linux/lib/libz.a  vcpkg_installed/x64-linux/lib/libpcre.a  -lm  -lpthread && :
/usr/bin/ld: vcpkg_installed/x64-linux/lib/libcrypto.a(libcrypto-lib-sha1_one.o): in function `SHA1':
sha1_one.c:(.text+0xa0): multiple definition of `SHA1'; vcpkg_installed/x64-linux/debug/lib/libldserverapi.a(sha1.c.o):/home/myname/deleteme/ld_error/vcpkg/buildtrees/launch-darkly-server/src/274bd6b472-130d2b5c68.clean/sha1.c:286: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
~~~

IIUC "clibs" is part of `launch-darkly-server`, and the primary definition of this symbol comes from `libcrypto.a` which is from port OpenSSL.
Or more correctly, the portfile addds "clibs" as a vendored dependency to "launch-darkly-server":
https://github.com/microsoft/vcpkg/blob/16ee2ecb31788c336ace8bb14c21801efb6836e4/ports/launch-darkly-server/portfile.cmake#L34-L40

If not simply switching to libcrypto, a macro might be another alternative to rename the binary symbol, with a single extra line.

--- a/ports/launch-darkly-server/fix-confilct-with-openssl.patch
+++ b/ports/launch-darkly-server/fix-confilct-with-openssl.patch
@@ -1,0 +1,26 @@
+diff --git a/sha1.c b/sha1.c
+index fe8da83..53ad2c4 100644
+--- a/sha1.c
++++ b/sha1.c
+@@ -279,7 +279,7 @@ void SHA1Final(
+     memset(&finalcount, '\0', sizeof(finalcount));
+ }
+ 
+-void SHA1(
++void SHA1_CLIBS(
+     char *hash_out,
+     const char *str,
+     int len)
+diff --git a/sha1.h b/sha1.h
+index 96bb008..53c4b01 100644
+--- a/sha1.h
++++ b/sha1.h
+@@ -36,7 +36,7 @@ void SHA1Final(
+     SHA1_CTX * context
+     );
+ 
+-void SHA1(
++void SHA1_CLIBS(
+     char *hash_out,
+     const char *str,
+     int len);

--- a/ports/launch-darkly-server/fix-depend-clib.patch
+++ b/ports/launch-darkly-server/fix-depend-clib.patch
@@ -1,0 +1,13 @@
+diff --git a/src/evaluate.c b/src/evaluate.c
+index f702419..e80fa6c 100644
+--- a/src/evaluate.c
++++ b/src/evaluate.c
+@@ -1246,7 +1246,7 @@ LDi_bucketUser(
+             char        digest[21], encoded[17];
+             const float longScale = 1152921504606846975.0;
+ 
+-            SHA1(digest, raw, strlen(raw));
++            SHA1_CLIBS(digest, raw, strlen(raw));
+ 
+             /* encodes to hex, and shortens, 16 characters in hex 8 bytes */
+             status = hexify(

--- a/ports/launch-darkly-server/portfile.cmake
+++ b/ports/launch-darkly-server/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
         findPCRE.patch
         FixStrictPrototypes.patch # required with clang-15
         removeWarningAsError.patch
+        fix-depend-clib.patch
 )
 
 
@@ -37,6 +38,8 @@ vcpkg_from_github(
     REF fa1d96ec293d2968791603548125e3274bd6b472
     SHA512 fd7dfbed4ac10e2c482da1cd460dabf0a53965e6fa17fab97156becb8214e435ee3605b2748705141380e254de7c32ab42da5e42cd6e4494f7ecaafb3b9e19f0
     HEAD_REF master
+    PATCHES
+        fix-confilct-with-openssl.patch
 )
 
 vcpkg_from_github(

--- a/ports/launch-darkly-server/vcpkg.json
+++ b/ports/launch-darkly-server/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "launch-darkly-server",
   "version": "2.8.6",
-  "port-version": 1,
+  "port-version": 2,
   "description": "LaunchDarkly server-side SDK for C/C++",
   "homepage": "https://github.com/launchdarkly/c-server-sdk",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3994,7 +3994,7 @@
     },
     "launch-darkly-server": {
       "baseline": "2.8.6",
-      "port-version": 1
+      "port-version": 2
     },
     "lazy-importer": {
       "baseline": "2023-08-03",

--- a/versions/l-/launch-darkly-server.json
+++ b/versions/l-/launch-darkly-server.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "cf4d27e890d7becddd87f89a3a0622199892ebd2",
+      "version": "2.8.6",
+      "port-version": 2
+    },
+    {
       "git-tree": "d2e41e5748cc5b77167205b83be64003e979e637",
       "version": "2.8.6",
       "port-version": 1

